### PR TITLE
Fixing probable bugs in core module

### PR DESCRIPTION
**Pull request by metodej**
fixes that should have been included in the first wave of fixes such as:
- added comments explaining removal of unnecessary check
- getValue() rewritten to valueAsString() or valueAsList()
- removed unnecessary condition and redundant assignments
- fixed test assertion

### CODE REVIEW

File: perun-core/src/test/java/cz/metacentrum/perun/core/entry/AttributesManagerEntryIntegrationTestAbstract.java [hunk 1]
```diff
@@ -912,10 +912,9 @@ public void getRichAttributesWithHoldersForAttributeDefinitionGetGroupFromUserAn
 	public void getAllSimilarAttributeNames() throws Exception {
 		System.out.println(CLASS_NAME + "getAllSimilarAttributeNames");
 
-		List<String> similarAttrNames = new ArrayList<>();
 		String name = "urn:perun:user:attribute-def:def:login-namespace";
-		similarAttrNames = perun.getAttributesManagerBl().getAllSimilarAttributeNames(sess, name);
-		//FIXME add assertion that tests the method
+		List<String> similarAttrNames = perun.getAttributesManagerBl().getAllSimilarAttributeNames(sess, name);
+		assertTrue("returned less than 0 names",similarAttrNames.size() > 0);
```
Comment (Vojtech-Sassmann) [inline perun-core/src/test/java/cz/metacentrum/perun/core/entry/AttributesManagerEntryIntegrationTestAbstract.java]:
```suggestion
		assertTrue("returned no names",similarAttrNames.isEmpty());
```

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_group_resource_attribute_def_def_projectDataLimit.java [hunk 2]
```diff
@@ -69,23 +69,22 @@ public void checkAttributeValue(PerunSessionImpl perunSession, Group group, Reso
...
-				projectDataLimitNumber = projectDataLimit.substring(numberMatcher.start(), numberMatcher.end());
-			} catch (IllegalStateException ex) {
-				projectDataLimitNumber = null;
-			}
-			try {
-				projectDataLimitLetter = projectDataLimit.substring(letterMatcher.start(), letterMatcher.end());
-			} catch (IllegalStateException ex) {
-				projectDataLimitLetter = "G";
-			}
+		projectDataLimit = (String) attribute.getValue();
```
Comment (Vojtech-Sassmann) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_group_resource_attribute_def_def_projectDataLimit.java]:
```suggestion
		projectDataLimit = attribute.valueAsString();
```

Comment (metodej) [reply perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_group_resource_attribute_def_def_projectDataLimit.java]: Ok, it should be fixed now.

File: perun-core/src/test/java/cz/metacentrum/perun/core/entry/AttributesManagerEntryIntegrationTestAbstract.java (hunk 1 above)
Comment (metodej) [reply perun-core/src/test/java/cz/metacentrum/perun/core/entry/AttributesManagerEntryIntegrationTestAbstract.java]: Ok, it should be changed now.

**Pull request merged**